contracts/forwarder/Eip712Forwarder.sol

```diff
@@ -64,47 +62,22 @@ contract Eip712Forwarder is IForwarder {
         nonces[req.from]++;
     }
 
-    /**
-     * Register a new Request typehash.
-     * @param typeName - the name of the request type.
-     * @param extraParams - params to add to the request type, after initial "_ForwardRequest request" param
-     * @param subTypes - subtypes used by the extraParams
-     * @param subTypes2 - more subtypes, if sorted after _ForwardRequest (e.g. if type starts with lowercase)
-     */
-    function registerRequestType(
-        string calldata typeName,
-        string calldata extraParams,
-        string calldata subTypes,
-        string calldata subTypes2
-    )
-    external
-    override
-    {
-        require(bytes(typeName).length > 0, "invalid typeName");
-        bytes memory types = abi.encodePacked(subTypes, GENERIC_TYPE, subTypes2);
-        string memory comma = bytes(extraParams).length > 0 ? "," : "";
-
-        bytes memory requestType = abi.encodePacked(
-            typeName, "(",
-            GENERIC_PARAMS, comma, extraParams, ")",
-            types
-        );
-        bytes32 requestTypehash = keccak256(bytes(requestType));
-        uint len = bytes(subTypes).length;
-        //sanity: avoid redefining our type, e.g.: subType="_ForwardRequest(whatever)_z"
-        require(len == 0 || bytes(subTypes)[len - 1] == ")", "invalid subType");
-        //sanity: parameters should not end parameters block
-        for (uint i = 0; i < bytes(extraParams).length; i++) {
-            require(bytes(extraParams)[i] != ")", "invalid extraParams");
+    function registerRequestType(string calldata typeName, string calldata typeSuffix) external override {
+
+        for (uint i = 0; i < bytes(typeName).length; i++) {
+            bytes1 c = bytes(typeName)[i];
+            require(c != "(" && c != ")", "invalid typename");
         }
+
+        bytes1 separator = ")";
+        if (bytes(typeSuffix).length > 0) separator = ",";
```

Comment:
Let's ban empty type suffix, and pre-register pure forwarder type in constructor. I don't like this conditional code in 'register' function.

Reply:
I don't really understand why you want to ban no-added-args.
The net result is that there is  exactly one allowed message name without extra parameters - and whatever name you like if you add a single field.
Why is that make sense?

Reply:
what does not make sense is code like ```if (unnecessaryCheck) { doUnnecessaryStuff() }``` in solidity. 
No extra parameters - no rename of a structure for you.

Want a custom name? Add field 'hello', set value to 'world'. I don't want to care about this use-case